### ISSUE TITLE
bug: `experimentalSlotFixes` (and `slotChildNodesFix` / `scopedSlotTextContentFix`) broken

### ISSUE THREAD

**Issue by johnjenkins** [ionic-team/stencil#6054]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

latest

### Current Behavior

Add 
```
extras: {
    experimentalSlotFixes: true,
    // or individually
    slotChildNodesFix: true,
   scopedSlotTextContentFix: true,
  },
 ```
 to `stencil.config`.
 
Then do something like `document.querySelector("my-component").childNodes` - where 'my-component' is `scoped: true`.
The nodelist returned will be the 'internal' nodes as rendered within the component.

Additionally, `scopedSlotTextContentFix` does not adhere to "return values described in [MDN's article on textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#description)" < as described in the Stencil documentation. 

### Expected Behavior

According to the documentation `document.querySelector("my-component").childNodes` 
 - where 'my-component' is `scoped: true` - shouljd *only* return the nodes that were slotted, not the internal, component rendered, nodes (and the same with `children` / slotted elements)
 
 `scopedSlotTextContentFix` should "return values described in [MDN's article on textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#description)" < as described in the Stencil documentation.
 

### Steps to Reproduce

add 
```
extras: {
    experimentalSlotFixes: true,
    // or individually
    slotChildNodesFix: true,
    scopedSlotTextContentFix: true,
  },
 ```
 to stencil.config, build components then do `.childNodes` / `.children` / `.textContent` on a `scoped: true` component

### Code Reproduction URL

https://codesandbox.io/p/devbox/stencil-web-component-forked-ds6rj3?workspaceId=72400648-5e20-4c72-af8a-d0ff6c9d8a44

**Comment by christian-bromann** [comment]
A fix for this was released in Stencil [`v4.3.0`](https://github.com/ionic-team/stencil/releases/tag/v4.23.0)

**Comment by christian-bromann** [comment]
A fix for this was released in Stencil [`v4.23.0`](https://github.com/ionic-team/stencil/releases/tag/v4.23.0)
